commands/proposal_submission.go

```diff
@@ -494,48 +516,47 @@ func checkNewTransferChanges(change *protoTypes.ProposalTerms_NewTransfer) Error
 	if changes.SourceType == protoTypes.AccountType_ACCOUNT_TYPE_UNSPECIFIED {
 		return errs.FinalAddForProperty("proposal_submission.terms.change.new_transfer.changes.source_type", ErrIsRequired)
 	}
-
+	validDest, ok := validTransfers[changes.SourceType]
 	// source account type may be one of the following:
-	if _, ok := validSources[changes.SourceType]; !ok {
+	if !ok {
 		return errs.FinalAddForProperty("proposal_submission.terms.change.new_transfer.changes.source_type", ErrIsNotValid)
 	}
-
 	if changes.DestinationType == protoTypes.AccountType_ACCOUNT_TYPE_UNSPECIFIED {
 		return errs.FinalAddForProperty("proposal_submission.terms.change.new_transfer.changes.destination_type", ErrIsRequired)
 	}
 
-	if changes.SourceType == protoTypes.AccountType_ACCOUNT_TYPE_GLOBAL_REWARD && changes.DestinationType == protoTypes.AccountType_ACCOUNT_TYPE_GLOBAL_REWARD {
+	if _, ok := validDest[changes.DestinationType]; !ok {
 		return errs.FinalAddForProperty("proposal_submission.terms.change.new_transfer.changes.destination_type", ErrIsNotValid)
 	}
+	dest := changes.DestinationType
 
-	if changes.SourceType == protoTypes.AccountType_ACCOUNT_TYPE_NETWORK_TREASURY && changes.DestinationType == protoTypes.AccountType_ACCOUNT_TYPE_NETWORK_TREASURY {
-		return errs.FinalAddForProperty("proposal_submission.terms.change.new_transfer.changes.destination_type", ErrIsNotValid)
-	}
-
-	// destination account type may be one of the following:
-	if _, ok := validDestinations[changes.DestinationType]; !ok {
-		return errs.FinalAddForProperty("proposal_submission.terms.change.new_transfer.changes.destination_type", ErrIsNotValid)
-	}
-
-	if changes.DestinationType == protoTypes.AccountType_ACCOUNT_TYPE_GENERAL && !IsVegaPublicKey(changes.Destination) {
+	// party accounts: check pubkey
+	if (dest == protoTypes.AccountType_ACCOUNT_TYPE_GENERAL ||
+		dest == protoTypes.AccountType_ACCOUNT_TYPE_BOND ||
+		dest == protoTypes.AccountType_ACCOUNT_TYPE_MARGIN) &&
+		!IsVegaPublicKey(changes.Destination) {
 		errs.AddForProperty("proposal_submission.terms.change.new_transfer.changes.destination", ErrShouldBeAValidVegaPublicKey)
 	}
 
-	if (changes.SourceType == protoTypes.AccountType_ACCOUNT_TYPE_GLOBAL_REWARD ||
-		changes.SourceType == protoTypes.AccountType_ACCOUNT_TYPE_NETWORK_TREASURY ||
-		changes.SourceType == protoTypes.AccountType_ACCOUNT_TYPE_GLOBAL_INSURANCE) &&
-		len(changes.Source) > 0 {
+	// insurance account type requires a source, other sources are global
+	if changes.SourceType == protoTypes.AccountType_ACCOUNT_TYPE_INSURANCE {
+		if len(changes.Source) == 0 {
+			return errs.FinalAddForProperty("proposal_submission.terms.change.new_transfer.changes.source", ErrIsNotValid)
+		}
+		// destination == source
+		if dest == changes.SourceType && changes.Source == changes.Destination {
```

Comment:
doesn't this apply to any source type?

Reply:
It does, but because of the map, I believe the only one where this check is actually required is market insurance pools